### PR TITLE
API documentation #780

### PR DESCRIPTION
**Pull request by cogutvalera**
PR for #780

### CODE REVIEW

**Comment by oxarbitrage** [comment]
nice job, ill take a look in more detail as soon as i can.

**Comment by cogutvalera** [comment]
Thank you very much ! Really appreciate all your efforts and help !

File: libraries/app/include/graphene/app/api.hpp [hunk 1]
```diff
@@ -176,10 +176,31 @@ namespace graphene { namespace app {
                                                                         uint32_t stop = 0,
                                                                         unsigned limit = 100,
                                                                         uint32_t start = 0) const;
-
+         /**
+          * @brief Get orders with b:a pair
+          * @param a Base in pair
+          * @param b Quote in pair
+          * @param limit Maximum number of orders to retrieve
+          * @return A list of orders with b:a pair
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: It returns history, not orders.

File: libraries/app/include/graphene/app/api.hpp [hunk 2]
```diff
@@ -176,10 +176,31 @@ namespace graphene { namespace app {
...
          vector<order_history_object> get_fill_order_history( asset_id_type a, asset_id_type b, uint32_t limit )const;
+
+         /**
+          * @brief Get markets with b:a pair
+          * @param a Base in pair
+          * @param b Quote in pair
+          * @param bucket_seconds Represents the length in seconds of bucket
+          * @param start Time in seconds from where buckets begin
+          * @param end Time in seconds where buckets stop
+          * @return A list of markets with b:a pair (maximum markets to retrieve is less than 200)
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: It returns history in one market, not markets. Descriptions of params seems off.

File: libraries/app/include/graphene/app/api.hpp [hunk 3]
```diff
@@ -343,8 +420,26 @@ namespace graphene { namespace app {
          asset_api(graphene::chain::database& db);
          ~asset_api();
 
+         /**
+          * @brief Get holders for a specific asset
+          * @param asset_id The specific asset
+          * @param start The start index of first asset
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: Seems wrong.

File: libraries/app/include/graphene/app/api.hpp [hunk 4]
```diff
@@ -343,8 +420,26 @@ namespace graphene { namespace app {
...
+          * @param limit Maximum limit must not exceed 100
+          * @return A holders for the specified asset
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: `A holders` sounds strange.

File: libraries/app/include/graphene/app/api.hpp [hunk 5]
```diff
@@ -176,10 +176,31 @@ namespace graphene { namespace app {
...
+          * @param start Time in seconds from where buckets begin
+          * @param end Time in seconds where buckets stop
+          * @return A list of markets with b:a pair (maximum markets to retrieve is less than 200)
+          */
          vector<bucket_object> get_market_history( asset_id_type a, asset_id_type b, uint32_t bucket_seconds,
                                                    fc::time_point_sec start, fc::time_point_sec end )const;
+
+         /**
+          * @brief Get tracked buckets
+          * @return A list of tracked buckets
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: Perhaps need to explain what's a bucket.

**Comment by cogutvalera** [comment]
Thank you very much ! Will improve soon ! Sorry for mistakes !

**Comment by cogutvalera** [comment]
improved, but there still maybe some wrong explanations, I will check myself again and will improve/fix what will find wrong.Thank you very much !

File: libraries/app/include/graphene/app/api.hpp [hunk 6]
```diff
@@ -325,12 +393,22 @@ namespace graphene { namespace app {
...
          verify_range_proof_rewind_result verify_range_proof_rewind( const blind_factor_type& nonce,
                                                                      const fc::ecc::commitment_type& commit, 
                                                                      const std::vector<char>& proof );
          
-                                         
+         /**
+          * @brief Get range proof info
+          * @param proof List of characters
+          * @return A range proof info structure with exponent, mantissa, min and max values
+          */
```
Comment (pmconrad) [inline libraries/app/include/graphene/app/api.hpp]: Do you understand how the crypto_api works?
Your documentation provides nothing useful that isn't obvious from the method/variable names.

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: not all details still researching this code in more details ... blind transactions are main idea for `crypto_api`

**Comment by cogutvalera** [comment]
I will improve this documentation more soon ...

**Comment by cogutvalera** [comment]
Guys, thanks for your time and efforts ! Please check my new changes when you will have enough time for it. Should I make any other changes in api documentation or it is ok for now ?

Thanks !

File: libraries/app/include/graphene/app/api.hpp [hunk 7]
```diff
@@ -343,8 +426,26 @@ namespace graphene { namespace app {
          asset_api(graphene::chain::database& db);
          ~asset_api();
 
+         /**
+          * @brief Get account holders for a specific asset
```
Comment (pmconrad) [inline libraries/app/include/graphene/app/api.hpp]: "account holders" doesn't make sense. Either "asset holders" or "accounts holding asset".

File: libraries/app/include/graphene/app/api.hpp [hunk 8]
```diff
@@ -343,8 +426,26 @@ namespace graphene { namespace app {
...
+          * @param asset_id The specific asset
+          * @param start The start index
+          * @param limit Maximum limit must not exceed 100
+          * @return A list of account holders for the specified asset
+          */
          vector<account_asset_balance> get_asset_holders( asset_id_type asset_id, uint32_t start, uint32_t limit  )const;
+
+         /**
+          * @brief Get account holders count for a specific asset
```
Comment (pmconrad) [inline libraries/app/include/graphene/app/api.hpp]: Same.

File: libraries/app/include/graphene/app/api.hpp [hunk 9]
```diff
@@ -176,10 +176,32 @@ namespace graphene { namespace app {
...
+          * @brief Get history of orders with b:a pair
+          * @param a Base in pair
+          * @param b Quote in pair
+          * @param limit Maximum number of orders to retrieve
+          * @return A history of orders with b:a pair, where a <= b, otherwise a and b will be swaped
+          */
          vector<order_history_object> get_fill_order_history( asset_id_type a, asset_id_type b, uint32_t limit )const;
+
+         /**
+          * @brief Get markets with b:a pair
```
Comment (pmconrad) [inline libraries/app/include/graphene/app/api.hpp]: Gets accumulated market history, not current markets

File: libraries/app/include/graphene/app/api.hpp [hunk 10]
```diff
@@ -176,10 +176,32 @@ namespace graphene { namespace app {
                                                                         uint32_t stop = 0,
                                                                         unsigned limit = 100,
                                                                         uint32_t start = 0) const;
-
+         /**
+          * @brief Get history of orders with b:a pair
+          * @param a Base in pair
+          * @param b Quote in pair
+          * @param limit Maximum number of orders to retrieve
+          * @return A history of orders with b:a pair, where a <= b, otherwise a and b will be swaped
```
Comment (pmconrad) [inline libraries/app/include/graphene/app/api.hpp]: If always a <= b then it doesn't make sense to call a "Base" and b "Quote". Backend is ignorant of UI market order.

File: libraries/app/include/graphene/app/api.hpp [hunk 11]
```diff
@@ -176,10 +176,32 @@ namespace graphene { namespace app {
...
          vector<order_history_object> get_fill_order_history( asset_id_type a, asset_id_type b, uint32_t limit )const;
+
+         /**
+          * @brief Get markets with b:a pair
+          * @param a Base in pair
+          * @param b Quote in pair
+          * @param bucket_seconds Represents the length in seconds of bucket
+          * @param start Time in seconds from where buckets begin
+          * @param end Time in seconds where buckets stop
+          * @return A history in market with b:a pair (maximum markets to retrieve is less than 200), where a <= b, otherwise a and b will be swaped
```
Comment (pmconrad) [inline libraries/app/include/graphene/app/api.hpp]: Same.

**Comment by pmconrad** [comment]
Your formatting changes seem to violate current conventions rather than improve things.

(Not that we're very consistent in that regard.)

**Comment by pmconrad** [comment]
Ok, keep your changes. Created #1318 for defining coding style.
@abitmore please re-review, I'm ok with the changes.

**Comment by cogutvalera** [comment]
Thanks !

File: libraries/app/include/graphene/app/api.hpp [hunk 12]
```diff
@@ -185,31 +185,29 @@ namespace graphene { namespace app {
                                                                         unsigned limit = 100,
                                                                         uint32_t start = 0) const;
          /**
-          * @brief Get history of orders with b:a pair
+          * @brief Get list of orders with b:a pair
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: Before you change the code, can you please explain why you think it's better to change this way? Is there other/better alternatives? Have you understood the purpose of the API? To me neither "history of orders" nor "list of orders" fits this API. Actually I think the word "orders" is not the best here.

File: libraries/app/include/graphene/app/api.hpp [hunk 13]
```diff
@@ -185,31 +185,29 @@ namespace graphene { namespace app {
...
           * @param a asset in pair
           * @param b asset in pair
-          * @param limit Maximum number of orders to retrieve
+          * @param limit Maximum limit of orders to retrieve
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: Same as above.

File: libraries/app/include/graphene/app/api.hpp [hunk 14]
```diff
@@ -185,31 +185,29 @@ namespace graphene { namespace app {
...
+          * @brief Get accumulated market history with b:a pair
+          * Thus supporting OHLC https://www.investopedia.com/terms/o/ohlcchart.asp
           * @param a asset in pair
           * @param b asset in pair
           * @param bucket_seconds Represents the length in seconds of bucket
-          * @param start Time in seconds from where buckets begin
-          * @param end Time in seconds where buckets stop
-          * @return A history in market with b:a pair (maximum markets to retrieve is less than 200), 
-          * where a <= b, otherwise a and b will be swaped
+          * @param start Time point in seconds since 1970 from where buckets begin
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: This description doesn't help much here.. You should put your feet in the user's shoes: what's the most helpful description when I'm looking at the document. Of course you need to understand the API in order to write document for it.

For this API, a user doesn't need to count how many seconds has been past since 1970.

IMHO it's better to provide an example E.G. "2018-09-12T18:00:00".

File: libraries/app/include/graphene/app/api.hpp [hunk 15]
```diff
@@ -185,31 +185,29 @@ namespace graphene { namespace app {
...
+          * @return A history in market with b:a pair (limit to retrieve is less than 200 buckets), where a less or equal than b
           */
          vector<bucket_object> get_market_history( asset_id_type a, asset_id_type b, uint32_t bucket_seconds,
                                                    fc::time_point_sec start, fc::time_point_sec end )const;
 
          /**
-          * @brief Get tracked buckets.
-          * Track market history by grouping orders into buckets of equal size measured in seconds 
-          * specified as a JSON array of numbers.
+          * @brief Get tracked groups of orders with equal size measured in seconds known as buckets
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: I'm Unsatisfied. The description is hard to understand.

File: libraries/app/include/graphene/app/api.hpp (hunk 12 above)
Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: maybe it's best fitted here to use "asset pairs" instead of "history/list of orders", because looking at code realization here https://github.com/bitshares/bitshares-core/blob/develop/libraries/app/api.cpp#L286-L309 I see that result is constructed with asset pairs a:b.
Am I wrong here or not ? Maybe I've missed something here ?

File: libraries/app/include/graphene/app/api.hpp (hunk 13 above)
Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: "Maximum count of asset pairs" instead of "Maximum limit of orders", IMHO its better for clear understanding. Am I wrong here or not ? Maybe I've missed something here ?

File: libraries/app/include/graphene/app/api.hpp (hunk 12 above)
Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: I think you still didn't get it. It's not your fault, because we lack many docs about the code here and elsewhere.

Please ask a trader how to describe/explain the API better.

>because looking at code ...

Please look deeper.

File: libraries/app/include/graphene/app/api.hpp [hunk 16]
```diff
@@ -185,11 +185,11 @@ namespace graphene { namespace app {
                                                                         unsigned limit = 100,
                                                                         uint32_t start = 0) const;
          /**
-          * @brief Get list of orders with b:a pair
+          * @brief Get asset pairs with b:a pair
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: "asset pairs" doesn't make sense. Please talk to traders to see what should be put here.

File: libraries/app/include/graphene/app/api.hpp (hunk 12 above)
Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: OK. Sure. Thanks !

File: libraries/app/include/graphene/app/api.hpp (hunk 16 above)
Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: OK. Sure. Thanks !

**Comment by cogutvalera** [comment]
> Jacob Walrus, [14.09.18 06:23]
> [In reply to Valera Cogut]
> "Fill order history" would probably sound better and be more obvious to the layperson what you are talking about if you simply used "market history". "Asset pairs" is pretty good as is but "trading pairs" might be better. "Order book" would make most sense to traders I think over "list of orders". Did that help answer your questions?

**Comment by cogutvalera** [comment]
IMHO for me as I understood correctly from Jacob Warlus answer the most makes sense here **"market history"**

File: libraries/app/include/graphene/app/api.hpp [hunk 17]
```diff
@@ -176,10 +184,32 @@ namespace graphene { namespace app {
...
+          * @param a asset in pair
+          * @param b asset in pair
+          * @param limit Maximum count of asset pairs to retrieve
+          * @return Asset pairs with b:a pair
+          */
          vector<order_history_object> get_fill_order_history( asset_id_type a, asset_id_type b, uint32_t limit )const;
+
+         /**
+          * @brief Get accumulated market history with b:a pair
+          * Thus supporting OHLC https://www.investopedia.com/terms/o/ohlcchart.asp
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: IMHO the link is not needed here. Traders know what's OHLC.

**Comment by abitmore** [comment]
@cogutvalera apparently you've learnt some. Please talk more with your friend.

File: libraries/app/include/graphene/app/api.hpp [hunk 18]
```diff
@@ -185,30 +185,31 @@ namespace graphene { namespace app {
                                                                         unsigned limit = 100,
                                                                         uint32_t start = 0) const;
          /**
-          * @brief Get asset pairs with b:a pair
+          * @brief Get order book with b:a pair
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: Absolutely wrong.

File: libraries/app/include/graphene/app/api.hpp [hunk 19]
```diff
@@ -185,30 +185,31 @@ namespace graphene { namespace app {
...
+          * @param end Time point in seconds where buckets stop, E.G. "YYYY-MM-DDThh:mm:ss"
           * @return A history in market with b:a pair (limit to retrieve is less than 200 buckets)
           */
          vector<bucket_object> get_market_history( asset_id_type a, asset_id_type b, uint32_t bucket_seconds,
                                                    fc::time_point_sec start, fc::time_point_sec end )const;
 
          /**
-          * @brief Get list of seconds for grouping asset pairs in OHLC https://en.wikipedia.org/wiki/Open-high-low-close_chart
-          * @return A list of seconds which are represented as unsigned integers
+          * @brief Get list of seconds for grouping a book's orders in OHLC
```
Comment (abitmore) [inline libraries/app/include/graphene/app/api.hpp]: Wrong as well.

File: libraries/app/include/graphene/app/api.hpp (hunk 18 above)
Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: What is better in your opinion ? Because trader told me this variant with `"order book"` is better, but it isn't. Can you please assist me just a bit ?

Thanks !

Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: An API querying for history won't return an order book, that's why.
Please try understand the API, then explain it to the trader and get feedback. Modifying code cluelessly is a waste of time.

Sure I can give my opinion or even write the document by myself, however, since you asked to work on this issue, you should try to get it done by yourself.

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: I've changed "order book" to "trading orders", is it wrong ?

Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: IMHO, "trading order" is a confusing phrase. Why not ask traders?

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: I've asked in DEX Trading group in Telegram all traders and just 1 answer got from Jacob Walrus who told me that "order book" is better choice.
Jacob Walrus is a founder of EasyDex as he told me:
"and just so you know who I am, I'm the founder (co-founder with my wife technically) of EasyDex"

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: IMHO as I thought since beginning **"fill ordrer"** name (the same as API call contains) is just exactly what we need, but tried to find more suitable keywords. So maybe **"fill order"** is the best choice, will research more this question, more deeply ...

Thanks !

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: After my researches within code API and different useful Trading info in web I've found 1 new solution-keyword: **"open warrant"** instead of "trading pairs", what do you think about it ? Isn't it confusing ?
```
@brief Get open warrants with b:a asset pairs
```
I think it makes more sense and is suitable here. What are your thoughts ?

Thanks !

Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: I've never seen such a word elsewhere, but I guess it's wrong. Seems you're still in the wrong direction. In order to show that you've understood the API, before you write document, can you please explain what this API is, in plain English?

Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: > Jacob Walrus who told me that "order book" is better choice.

I guess you asked wrong question to him or in the channel, thus got a unrelated answer.

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: the main idea of `get_fill_order_history` API method is to retrieve a vector (limited in size) of `fill_order_operations` with b:a asset pairs.

Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: Correct. Then what does a "fill_order_operation" object mean?

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: The purpose of `fill_order_operation` is tracking account history, also it is a `virtual` operation.

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: here are all members of `fill_order_operation`:

```
      object_id_type      order_id;
      account_id_type     account_id;
      asset               pays;
      asset               receives;
      asset               fee; // paid by receiving account
      price               fill_price;
      bool                is_maker;
```

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: so `fill_order_operation` is needed when we want to place a trading order, it is required to fill the form on either the ask or the bid side (respectively, buy or sell side). You will need to define a price and an amount to sell/buy. The cost for this order will be calculated automatically.

Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: Check when will a `fill_order_operation` be created.

Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: In addition, what is a "virtual" operation.

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: `virtaul operations` are those operations (derived from `base_operation` also) which are produced by blockchain itself, thus means they are not triggered by users.

Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: > they are not directly triggered by user operations

Not exactly.

Why the blockchain itself "produces" such operations?

Why this API is querying for such operations?

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: **`fill_order_operation`** is created just in next 3 places:
1. **`fill_limit_order`** https://github.com/bitshares/bitshares-core/blob/develop/libraries/chain/db_market.cpp#L782
2. **`fill_call_order`** https://github.com/bitshares/bitshares-core/blob/develop/libraries/chain/db_market.cpp#L866
3. **`fill_settle_order`** https://github.com/bitshares/bitshares-core/blob/develop/libraries/chain/db_market.cpp#L894

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: @abitmore Thank you for help ! Your questions helped me a lot to understood more details in correct way/direction. I want to research more deeply the code and especially real & virtual operations. Also I want to answer on your 2 questions:

> Why the blockchain itself "produces" such operations?
> 
> Why this API is querying for such operations?

1. For matching orders
2. For tracking account history and to speed up `replay` process

If you have any other questions, please just ask me and I will try to answer on all them.

Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: Both answers are not accurate, although some comments in the code said so, the comments are outdated.

Virtual operations can be seen as (a subset of) results of processing other operations or processing post-transaction chain logic. `fill_order_operation` objects are NOT "for matching orders", instead, they're results of "matching orders". Why the blockchain produced them? For easier tracking. A trader wants to know what's the result when he placed an order, so every time his order is filled (understand what is `fill`?), a `fill_order_operation` object is created, when the trade queries for his account history, he can see the results. It has nothing to do with replay speed.

However, in this issue we're talking about market history but not account history. Now can you explain it?

**Comment by cogutvalera** [comment]
just for the info: pushed new commit **"fixed: (trading orders) -> (matching orders)"**

File: libraries/app/include/graphene/app/api.hpp (hunk 18 above)
Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: According to the link in your comment:
> Matching orders is the process ...

The API doesn't return a "process" or list of "process".

Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: Investopedia IS correct. However, to fit our doc, you need to adjust the wording according to English grammar.

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: I thought **"matching orders"** is `stable phrase` in trader's slang

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: after our discussion and a lot of efforts trying to figure out the best suitable solution with @abitmore about API doc, `fill_order_operation` and after my code researches I've some results for your judgement:

1. **"matching orders"** can be filled partially, so traders will be confused with such API doc description (@abitmore helped me here with more details & examples, thanks a lot !)
2. this API method `get_fill_order_history` will return `filled orders` which also are `confirmed` by blockchain, so the best solution is to use **"confirmed orders"**, thus traders will understand completely what will be returned to them.

Thanks !

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: IMHO perhaps next:
1. just simply **"filled orders"**
2. or combination **"filled and confirmed orders"**

2nd version is more accurate and descriptive IMHO

Comment (abitmore) [reply libraries/app/include/graphene/app/api.hpp]: "filled orders" are very similar to "matched orders", and I've said why it's not good use them as is (note: I didn't say they can't be used in a sentence that's clearer). "confirmed" is very confusing. "filled and confirmed" doesn't make sense IMHO.

Can you **reword** the whole sentence? It's not only about choosing a noun or a phrase, it's the whole sentence that should be clearer. If unable to clearly describe it in 6 words or one sentence, please use more words and more sentences.

Comment (cogutvalera) [reply libraries/app/include/graphene/app/api.hpp]: ok, understood, sure. Thanks !

**Comment by cogutvalera** [comment]
changed last commit's message

**Comment by abitmore** [comment]
I give up. @cogutvalera I think it's better to update the docs about orders as follows (please wrap long lines):
```
         /**
          * @brief Get details of order executions occurred most recently in a trading pair
          * @param a One asset in a trading pair
          * @param b The other asset in the trading pair
          * @param limit Maximum records to return
          * @return a list of order_history objects, in "most recent first" order
          */
         vector<order_history_object> get_fill_order_history( asset_id_type a, asset_id_type b, uint32_t limit )const;

         /**
          * @brief Get OHLCV data of a trading pair in a time range
          * @param a One asset in a trading pair
          * @param b The other asset in the trading pair
          * @param bucket_seconds Length of each time bucket in seconds. Note: it need to be within result of get_market_history_buckets() API, otherwise no data will be returned
          * @param start The start of a time range, E.G. "2018-01-01T00:00:00"
          * @param end The end of the time range
          * @return A list of OHLCV data, in "least recent first" order. If there are more than 200 records in the specified time range, the first 200 records will be returned.
          */
         vector<bucket_object> get_market_history( asset_id_type a, asset_id_type b, uint32_t bucket_seconds,
                                                   fc::time_point_sec start, fc::time_point_sec end )const;

         /**
          * @brief Get OHLCV time bucket lengths supported (configured) by this API server
          * @return A list of time bucket lengths in seconds. E.G. if the result contains a number "300", it means this API server supports OHLCV data aggregated in 5-minute buckets.
          */                                          
         flat_set<uint32_t> get_market_history_buckets()const;
```

References:
* https://www.investopedia.com/terms/e/execution.asp
* https://www.investopedia.com/terms/f/fill.asp

**Comment by cogutvalera** [comment]
@abitmore Thanks ! Changes are done.

**Comment by cogutvalera** [comment]
Thanks !
